src/main/java/org/reactome/qa/doisuggester/Main.java

```diff
@@ -301,64 +298,74 @@ private static void printOutput(Map<GKInstance, Set<Map<GKInstance, GKInstance>>
 
 	private static boolean hasNewInstanceEdit(List<GKInstance> currentRLEInstanceEdits, List<GKInstance> previousRLEInstanceEdits)
 	{
+		// start by assuming false and set to true when conditions require it.
+		boolean newInstanceEdit = false;
+		// I guess Justin assumed that if there are more InstanceEdits of the current version of an object, the it *must* be new.
+		// This is *probably* a safe assumption to make.
 		if (currentRLEInstanceEdits.size() != previousRLEInstanceEdits.size())
 		{
-			return true;
+			newInstanceEdit = true;
 		}
 		else
 		{
-			for (int i = 0; i < currentRLEInstanceEdits.size(); i++)
+			int i = 0;
+			while (i < currentRLEInstanceEdits.size() && !newInstanceEdit)
 			{
 				GKInstance currentIE = currentRLEInstanceEdits.get(i);
 				GKInstance previousIE = previousRLEInstanceEdits.get(i);
+				// It seems that in the case where current and previous versions have the same
+				// number of InstanceEdits, comparing DBIDs indicates if an InstanceEdit is new.
+				// I'm assuming Justin tested this theory more rigorously.
 				if (!currentIE.getDBID().equals(previousIE.getDBID()))
 				{
-					return true;
+					newInstanceEdit = true;
 				}
+				i++;
```

Comment:
This also seems dangerous.

If the number of instance edits is the same in both databases, a difference in db id would indicate one instance edit or more got removed and the same number were added.  Instance edits shouldn't disappear, so I would think this should log/trigger an error?

This also assumes the order of the instance edits in both databases are the same.  They almost for sure will be, but if all the instance edits are actually the same but in a different order then it would indicate a new instance edit is there because each pair compared in order would have different db ids.

Reply:
Also good points that can be considered for future PRs.